router.go

```diff
@@ -581,12 +588,22 @@ func (p *ControllerRegister) geturl(t *Tree, url, controllName, methodName strin
 }
 
 func (p *ControllerRegister) execFilter(context *beecontext.Context, urlPath string, pos int) (started bool) {
+	var preFilterParams map[string]string
 	for _, filterR := range p.filters[pos] {
 		if filterR.returnOnOutput && context.ResponseWriter.Started {
 			return true
 		}
+		if filterR.resetParams {
+			preFilterParams = context.Input.Params()
+		}
 		if ok := filterR.ValidRouter(urlPath, context); ok {
 			filterR.filterFunc(context)
+			if filterR.resetParams {
+				context.Input.Reset(context)
```

Comment:
`Input.Reset` has a log impact. while you only recover the params, but the Reset will reset all the data:

```
input.CruSession = nil
input.pnames = input.pnames[:0]
input.pvalues = input.pvalues[:0]
input.data = nil
input.RequestBody = []byte{}
```